service/resource/persistentvolume/resource.go

```diff
@@ -133,3 +136,100 @@ func pvToRecyclePV(v interface{}) (*RecyclePersistentVolume, error) {
 
 	return rpv, nil
 }
+
+// newRecycleStateAnnotation create new PersistentVolume object with
+// updated recycle state annotation.
+func (r *Resource) newRecycleStateAnnotation(pv *apiv1.PersistentVolume, recycleAnnotation string) (*apiv1.PersistentVolume, error) {
+
+	updatedpv := &apiv1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        pv.Name,
+			Annotations: pv.Annotations,
+			Labels: pv.Labels,
+		},
+		Spec: apiv1.PersistentVolumeSpec{
+			Capacity:                      pv.Spec.Capacity,
+			StorageClassName:              pv.Spec.StorageClassName,
+			AccessModes:                   pv.Spec.AccessModes,
+			PersistentVolumeReclaimPolicy: pv.Spec.PersistentVolumeReclaimPolicy,
+			PersistentVolumeSource:        pv.Spec.PersistentVolumeSource,
+		},
+	}
```

Comment:
Maybe dumb question but can you create a copy of pv? All the values are the same except the annotation which is set later.

Reply:
That is a good question. @kopiczko asked it as well. And I tried DeepCopy, which is okay for updates. But there is also the case, where I need delete persistent volume and create the new one with the same fields. And that doesn't work as new volume can't have all that metadata, which existing volume has (uid, resourceVersion, selflink etc).

Reply:
Got it. Makes sense for the delete case.